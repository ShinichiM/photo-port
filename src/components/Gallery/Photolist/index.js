import React, { useState } from 'react';
import { getCategories } from './gallery-categories';
import Modal from '../../Modal';

function Photolist({ category }) {
    const [photos] = useState(getCategories());
    const [currentPhoto, setCurrentPhoto] = useState();
    const [isModalOpen, setIsModalOpen] = useState(false);

    const currentPhotos = photos.filter((photo) => photo.category === category);

    function toggleModal(photo, index) {
        setCurrentPhoto({...photo, index: index});
        setIsModalOpen(!isModalOpen);
    };

    return(
        <div className='flex-row'>
        {isModalOpen && <Modal currentPhoto={currentPhoto} onClose={toggleModal}></Modal>}
            {currentPhotos.map((photo, index) => (
                <img 
                src={require(`../../../assets/small/${category}/${index}.jpg`)} 
                className='img-thumbnail mx-1'
                alt={photo.name}
                key={photo.name}
                onClick={() => toggleModal(photo, index)}
                >
                </img>
            ))}
        </div>
    )
};

export default Photolist;