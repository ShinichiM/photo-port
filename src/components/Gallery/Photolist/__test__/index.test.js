import React from 'react';
import { render, cleanup } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import Photolist from '..';

afterEach(cleanup);

describe('Photolist Testing', () => {
    const category = {
        name: 'portrait',
        description: 'Portraits of animals'
    };

    it('renders', () => {
        render(<Photolist></Photolist>);
    });
    it('snapshot', () => {
        const { asFragment } = render(<Photolist></Photolist>);
        expect(asFragment()).toMatchSnapshot();
    });
});