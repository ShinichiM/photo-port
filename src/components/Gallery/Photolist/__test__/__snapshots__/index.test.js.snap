// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Photolist Testing snapshot 1`] = `
<DocumentFragment>
  <div
    class="flex-row"
  />
</DocumentFragment>
`;
